src/ja/nicomanga/src/eu/kanade/tachiyomi/extension/ja/nicomanga/Nicomanga.kt

```diff
@@ -0,0 +1,156 @@
+
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/manga-list.html?page=$page&sort=views&sort_type=DESC")
+
+    override fun searchMangaParse(response: Response): MangasPage = mangaListParse(response)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = GET("$baseUrl/manga-list.html?page=$page&artist=&author=&group=&m_status=&name=$query&genre=&ungenre=&magazine=&sort=last_update&sort_type=DESC")
+
+    override fun mangaDetailsParse(response: Response): SManga = SManga.create().apply {
+        val doc = Jsoup.parse(response.body.string())
+        author = doc.select("ul.manga-info a[href^=\"manga-author\"]").joinToString { it.text() }
+        genre = doc.select("ul.manga-info a[href^=\"manga-list-genre\"]").joinToString { it.text() }
+        val statusText = statusRegex.find(doc.select(".manga-info li:has(i.fa-spinner) a").attr("href"))?.groupValues?.get(0) ?: ""
+        status = when (statusText) {
+            "on-going" -> {
+                SManga.ONGOING
+            }
+            "completed" -> {
+                SManga.COMPLETED
+            }
+            else -> {
+                SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = Jsoup.parse(response.body.string())
+        val chapters = ArrayList<SChapter>()
+        val chapterList = doc.select("ul > a")
+        val mangaId = response.request.url.toUrl().toString().substringAfterLast('/').substringBeforeLast('.')
+        val sharedPref = application.getSharedPreferences("source_${id}_time_found:$mangaId", 0)
+        val editor = sharedPref.edit()
+        var lastNum = 0f
+        for (chapter in chapterList) {
+            chapters.add(
+                SChapter.create().apply {
+                    name = chapter.attr("title").trim()
+                    setUrlWithoutDomain(URL(URL(baseUrl), chapter.attr("href")).toString())
+                    chapter_number = floatRegex.find(chapter.attr("title").trim())?.groupValues?.get(0)?.toFloat() ?: (lastNum + 0.01f)
+                    lastNum = chapter_number
+                    val dateFound = System.currentTimeMillis()
+                    if (!sharedPref.contains(chapter_number.toString())) {
+                        editor.putLong(chapter_number.toString(), dateFound)
+                    }
+                    date_upload = sharedPref.getLong(chapter_number.toString(), dateFound)
+                },
+            )
+        }
+        editor.apply()
+        return chapters
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val slug = urlRegex.find(manga.url)?.groupValues?.get(0) ?: ""
+        return GET("$baseUrl/app/manga/controllers/cont.Listchapterapi.php?slug=$slug")
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val doc = Jsoup.parse(response.body.string())
```

Comment:
this is usually an indication you should use `ParsedHttpSource`, so you can override `pageListParse(document: Document)` etc.